### ISSUE TITLE
Exceptional.publish seems broken?

### ISSUE THREAD

**Issue by bryanmills** [jzempel/flask-exceptional#10]
I'm trying to integrate flask-exceptional into our rq job queue. This seemed straightforward since you have provided Exceptional.publish but I've ran into a few issues. I'm not sure if this is user error or an actual regression. It appears that the flask-extension code is still expecting the request context when it shouldn't.

Here is my error handler code for rq:

``` python
def report_to_exceptional_handler(job, exc_type, exc_value, traceback):                                                                                    
    tb = Traceback(exc_type, exc_value, traceback)                                                                                                                                                                                                                                                                   
    Exceptional.publish(gotham.app.config, tb)  
```

I then get the following exception.

```
[2013-02-07 21:16] DEBUG: horse: Invoking exception handler <function report_to_exceptional_handler at 0xb7e52a8>
Traceback (most recent call last):
  File "worker.py", line 31, in <module>
    w.work()
  File "/home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/rq/worker.py", line 326, in work
    self.fork_and_perform_job(job)
  File "/home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/rq/worker.py", line 342, in fork_and_perform_job
    self.main_work_horse(job)
  File "/home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/rq/worker.py", line 377, in main_work_horse
    success = self.perform_job(job)
  File "/home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/rq/worker.py", line 402, in perform_job
    self.handle_exception(job, *sys.exc_info())
  File "/home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/rq/worker.py", line 443, in handle_exception
    fallthrough = handler(job, *exc_info)
  File "worker.py", line 16, in report_to_exceptional_handler
    Exceptional.publish(gotham.app.config, tb)
  File "/home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/flask_exceptional.py", line 144, in publish
    exceptional._post_data(traceback=traceback)
  File "/home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/flask_exceptional.py", line 232, in _post_data
    app = context.app if context else stack.top.app
AttributeError: 'NoneType' object has no attribute 'app'
```

It appears that this code is trying to pull out the variable stack.app.top where stack is really a flask._request_ctx_stack. However when I'm in my job queue we do not have a request context which is exactly what I thought the publish function was built to handle. Am I just using this incorrectly or should I submit a patch?

Thanks.

-bryan

**Comment by bryanmills** [comment]
Going on the assumption that this is a bug here is my quick patch that fixes the issue.

``` patch
--- flask_exceptional.py    2013-02-07 16:50:28.155515384 -0500
+++ /home/bryan/.virtualenvs/gotham/local/lib/python2.7/site-packages/flask_exceptional.py  2013-02-07 16:42:00.358890780 -0500
@@ -1,3 +1,4 @@
+
 # -*- coding: utf-8 -*-
 """
     flask_exceptional
@@ -140,7 +141,7 @@
             app.config.from_object(config)

         exceptional.init_app(app)
-        exceptional._post_data(traceback=traceback)
+        exceptional._post_data(traceback=traceback, app=app)

     @staticmethod
     def test(config):
@@ -219,7 +220,7 @@

         return ret_val

-    def _post_data(self, context=None, traceback=None):
+    def _post_data(self, context=None, traceback=None, app=None):
         """POST data to the the Exceptional API. If DEBUG is True then data is
         sent to ``EXCEPTIONAL_DEBUG_URL`` if it has been defined. If TESTING is
         true, error data is stored in the global ``flask.g.exceptional``
@@ -228,7 +229,8 @@
         :param context: Default ``None``. The current application context.
         :param traceback: Default ``None``. The exception stack trace.
         """
-        app = context.app if context else stack.top.app
+        if app is None:
+            app = context.app if context else stack.top.app
         application_data = self.__get_application_data(app)
         client_data = {
             "name": "flask-exceptional",
```

**Comment by jzempel** [comment]
@bryanmills - ack! I had my GitHub notifications turned down and didn't see this until now. It looks like you've got the right patch here. I'll incorporate it and get a new version pushed.

**Comment by jzempel** [comment]
Fixed by f87e9464e6cc304f4a46c98ef7aeacdc89905221. Uploaded [version 0.5.3](https://pypi.python.org/pypi/Flask-Exceptional/0.5.3) to PyPI.
